oceanprotocol/market
# graphql and onchain storage support

- Fixes #1724 .

Changes proposed in this PR:

- Adding GraphQL integration
- Adding OnChain data integration

Current state:
<img width="911" alt="Screenshot 2022-12-08 at 09 53 16" src="https://user-images.githubusercontent.com/13741335/206464404-46733c31-2e7f-45e3-bd4d-59ccff027665.png">

To do: 
- [x] merge IPFS / Arw integration
- [x] change textarea component
-- [x] install codemirror
-- [x] integrate codemirror into component
-- [x] style codemirror for dark/light + small fixes
-- [x] change layout & validation component
-- ~~[ ] find a way to validate graphql query without schema~~
-- [x] create headers component
-- [x] integrate headers on graphql?
-- [x] integrate headers on url?
-- [x] uncomment headers to provider calls? #1554 
- [x] connect both storages to provider
- [x] finish validation
- [x] add to publish form 
- [x] add to edit form 
- [x] fix preview edit form 
- [x] review tooltip text
- [x] fix tab selection when step changes
  - [x] test publish / edit / view
  - [x] test download (free and fixed) / consume / C2D 


______


### Testing User Flow:

Publish (Free and Fixed price):
- publish assets with GraphQL queries (test more than 1) without headers
- publish assets with GraphQL query (test more than 1) with headers (you can use: https://dgraph.io)
- publish assets with SM (smart contract) abi (test more than 1, with different response type)

Asset Details
- all assets should be correctly detailed (check specially the file informacion)

Edit Asset:
- edit storage type of some of your assets. If you published with GraphQL, than change it any of the other and make sure it's all working as expected

Consume
- consume and download all assets

Compute
- compute what published previously


_____
